fe/fe-core/src/main/java/org/apache/doris/analysis/AnalyzeStmt.java

```diff
@@ -166,46 +150,81 @@ private void checkAnalyzePriv(String dbName, String tblName) throws AnalysisExce
     }
 
     private void checkPartitionNames() throws AnalysisException {
-        if (optPartitionNames != null) {
-            optPartitionNames.analyze(analyzer);
-            if (tableName != null) {
-                Database db = analyzer.getEnv().getInternalCatalog().getDbOrAnalysisException(tableName.getDb());
-                OlapTable olapTable = (OlapTable) db.getTableOrAnalysisException(tableName.getTbl());
-                if (!olapTable.isPartitioned()) {
-                    throw new AnalysisException("Not a partitioned table: " + olapTable.getName());
-                }
-                List<String> names = optPartitionNames.getPartitionNames();
-                Set<String> olapPartitionNames = olapTable.getPartitionNames();
-                List<String> tempPartitionNames = olapTable.getTempPartitions().stream()
-                        .map(Partition::getName).collect(Collectors.toList());
-                Optional<String> optional = names.stream()
-                        .filter(name -> (tempPartitionNames.contains(name)
-                                || !olapPartitionNames.contains(name)))
-                        .findFirst();
-                if (optional.isPresent()) {
-                    throw new AnalysisException("Temporary partition or partition does not exist");
-                }
-            } else {
-                throw new AnalysisException("Specify partition should specify table name as well");
+        if (partitionNames != null) {
+            partitionNames.analyze(analyzer);
+            Database db = analyzer.getEnv().getInternalCatalog()
+                    .getDbOrAnalysisException(tableName.getDb());
+            OlapTable olapTable = (OlapTable) db.getTableOrAnalysisException(tableName.getTbl());
+            if (!olapTable.isPartitioned()) {
+                throw new AnalysisException("Not a partitioned table: " + olapTable.getName());
+            }
+            List<String> names = partitionNames.getPartitionNames();
+            Set<String> olapPartitionNames = olapTable.getPartitionNames();
+            List<String> tempPartitionNames = olapTable.getTempPartitions().stream()
+                    .map(Partition::getName).collect(Collectors.toList());
+            Optional<String> optional = names.stream()
+                    .filter(name -> (tempPartitionNames.contains(name)
+                            || !olapPartitionNames.contains(name)))
+                    .findFirst();
+            if (optional.isPresent()) {
+                throw new AnalysisException("Temporary partition or partition does not exist");
```

Comment:
The exception message seems not match the computation logic  of `optional` variable